### ISSUE TITLE
Problem installing 2decomp : -DMPIEXEC_MAX_NUMPROCS

### ISSUE THREAD

**Issue by LionelGelebart** [2decomp-fft/2decomp-fft#297]
### Discussed in https://github.com/2decomp-fft/2decomp-fft/discussions/296

When I use cmake with option -DMPIEXEC_MAX_NUMPROCS : 
the variable MAX_NUMPROCS is modified instead of MPIEXEC_MAX_NUMPROCS

Do observe the same thing?

**Comment by pbartholomew08** [comment]
You are right, I can reproduce this - I've always used the `ccmake` interface to configure this so never noticed.

If you look at the github actions, they follow this pattern:
```
cmake -B build -S . <general options> -DBUILD_TESTING=ON
# You could optionally build here

# Configure, build and run the tests for parallel execution
cmake -B build -S . -DMPIEXEC_MAX_NUM_PROCS=<NP>
cmake --build buil
ctest --test-dir build
```
I've ran the 2 configure steps and I can see that `MPIEXEC_MAX_NUM_PROCS` is modified correctly following this. I think this behaviour is related to how cmake caches variables @rfj82982?

**Comment by pbartholomew08** [comment]
however I should emphasise this only affects how the tests are run using `ctest` - you can manually run them using `mpirun -np <np> ...` and codes using the library will be unaffected by these settings.

**Comment by rfj82982** [comment]
I had a better look at this only now. Sorry for the delay, just came back from the US this morning. 
`-DMPIEXEC_MAX_NUMPROCS=XX` works only with `ccmake` or when the configure stage is re-executed a second time (as in our CI with the finer meshes tests). We had to force `CMake` to run `ctest` with one MPI rank because of our GPU implementation. By default `CMake` run the tests using the maximum available number of cores on the system, but if the number of GPU is smaller (which is always the case) our tests are going to fail. This is a consequence of our GPU strategy where we associate 1 GPU per 1 MPI ranks. Forcing to run the test in serial will make sure that `ctest` for GPU won't fail. If people want to run `ctest` with more cores this can be enforced calling again the configure step. I think we have to write this somewhere in the README or INSTALL

**Comment by mathrack** [comment]
Re-opening. This issue should be closed when the README or INSTALL is updated as mentioned above.

**Comment by mathrack** [comment]
All the tests are using the following syntax :
```
if (BUILD_TARGET MATCHES "gpu")
  file(COPY bind.sh DESTINATION ${run_dir})
  add_test(NAME test2d COMMAND ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${MPIEXEC_MAX_NUMPROCS} ./bind.sh $<TARGET_FILE:test2d> ${TEST_ARGUMENTS} WORKING_DIRECTORY ${run_dir})
else ()
  add_test(NAME test2d COMMAND ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${MPIEXEC_MAX_NUMPROCS} $<TARGET_FILE:test2d> ${TEST_ARGUMENTS} WORKING_DIRECTORY ${run_dir})
endif ()
```
Could we fix the present issue with something like
```
if (BUILD_TARGET MATCHES "gpu")
  file(COPY bind.sh DESTINATION ${run_dir})
  add_test(NAME test2d COMMAND ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${MPIEXEC_NGPUS} ./bind.sh $<TARGET_FILE:test2d> ${TEST_ARGUMENTS} WORKING_DIRECTORY ${run_dir})
else ()
  add_test(NAME test2d COMMAND ${MPIEXEC_EXECUTABLE} ${MPIEXEC_NUMPROC_FLAG} ${MPIEXEC_MAX_NUMPROCS} $<TARGET_FILE:test2d> ${TEST_ARGUMENTS} WORKING_DIRECTORY ${run_dir})
endif ()
```
The cmake variable `MPIEXEC_NGPUS` would be set to one by default.

**Comment by rfj82982** [comment]
This issue has been resolved with PR #378
